internal/events/api/event.go

```diff
@@ -72,3 +72,7 @@ type InfoEvent interface {
 	BaseEvent
 	GetInfo() string
 }
+
+type Uploader interface {
```

Comment:
Do we intend on creating a mock or alternate implementation for this?

Is there any other reason we have this interface?

Reply:
Thought about creating a mock for unit tests, but maybe there's no need to. What do you think?

Reply:
If you need a mock for unit tests then it's fine, but I just mention this when I see an interface without multiple concrete implementations.

Reply:
I see, thanks! I think I'll add a mock for unit testing, but remove it if I realize a mock isn't needed.

Reply:
Also I would put this with the implementation of the interface rather than in the events package unless there's some reason for it to be here that I'm not seeing.

Reply:
Thanks for commenting on this. I agree that it doesn't fit with the events items. I moved this to its own folder with the interface as a [separate file](https://github.com/openshift/assisted-service/pull/4880/files#diff-97008778ca8858564f55e9743e204f1f2a4a6475a2fe58517fdbff858d4e8e5aR1). What do you think?